### PR TITLE
[Bug][Connector-v2][RedisSink]Fix failures caused by different cluster mode operations.

### PR DESCRIPTION
**Pull request by lightzhao**
## Purpose of this pull request
If the redis mode is a cluster type, it will fail when operating the expire method.
So different redis cluster types should declare and use corresponding objects.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — DISMISSED** [review]
Thanks @lightzhao , over all fine. @TyrantLucifer maybe you should take a look, please.

**Comment by Hisoka-X** [comment]
The code style should be fixed.

File: seatunnel-connectors-v2/connector-file/pom.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
         <module>connector-file-base-hadoop</module>
         <module>connector-file-sftp</module>
         <module>connector-file-s3</module>
-        <module>connector-file-jindo-oss</module>
+        <!--        <module>connector-file-jindo-oss</module>-->
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/pom.xml]: Revert

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/pom.xml]: sorry. revert finsih.

**Comment by zhilinli123** [comment]
hello @lightzhao   Can you provide v2 connector documentation for this expressing thanks

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Expire feature has some problems, I think this feature https://github.com/apache/seatunnel/pull/4975 should revert and redesign it. For expire  key, we should only operation once not every time after set operation.

**Comment by TyrantLucifer** [comment]
BTW, whether in cluster mode or standalone mode, all operations should be abstracted into upper level interfaces

**Comment by lightzhao** [comment]
> Expire feature has some problems, I think this feature #4975 should revert and redesign it. For expire key, we should only operation once not every time after set operation.

What does it mean to operate only once, and how to implement only one operation?

**Comment by lightzhao** [comment]
> BTW, whether in cluster mode or standalone mode, all operations should be abstracted into upper level interfaces

It can be designed like this, but the repetition of code will be very high.

**Comment by lightzhao** [comment]
> BTW, whether in cluster mode or standalone mode, all operations should be abstracted into upper level interfaces

This looks good, abstract class + interface can implement different client operations.

**Comment by lightzhao** [comment]
> hello @lightzhao Can you provide v2 connector documentation for this expressing thanks

ok, i try it.

**Comment by lightzhao** [comment]
I will redesign.

**Pull request closed without merging**